tasks/EchoesToFrameConverterTask.cpp

```diff
@@ -0,0 +1,151 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "EchoesToFrameConverterTask.hpp"
+#include <base-logging/Logging.hpp>
+#include <base/samples/RigidBodyState.hpp>
+#include <math.h>
+#include <opencv2/imgproc/imgproc.hpp>
+
+using namespace radar_base;
+using namespace cv;
+using namespace base::samples::frame;
+EchoesToFrameConverterTask::EchoesToFrameConverterTask(std::string const& name)
+    : EchoesToFrameConverterTaskBase(name)
+{
+}
+
+EchoesToFrameConverterTask::~EchoesToFrameConverterTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See EchoesToFrameConverterTask.hpp for more detailed
+// documentation about them.
+
+bool EchoesToFrameConverterTask::configureHook()
+{
+    if (!EchoesToFrameConverterTaskBase::configureHook())
+        return false;
+    configureOutput();
+    updateLookUpTable();
+    return true;
+}
+bool EchoesToFrameConverterTask::startHook()
+{
+    if (!EchoesToFrameConverterTaskBase::startHook())
+        return false;
+    return true;
+    m_last_sample = base::Time::now();
+}
+void EchoesToFrameConverterTask::updateHook()
+{
+    EchoesToFrameConverterTaskBase::updateHook();
+
+    Radar radar_echo;
+    if (_echo.read(radar_echo, false) == RTT::NewData) {
+        base::Angle yaw_correction = base::Angle::fromRad(0);
+        base::samples::RigidBodyState sensor2ref_pose;
+        if (_sensor2ref_pose.read(sensor2ref_pose) != RTT::NoData) {
+            yaw_correction = base::Angle::fromRad(sensor2ref_pose.getYaw());
+        }
+
+        do {
+            if (m_current_sweep_size != radar_echo.sweep_length ||
+                m_current_num_angles != (2 * M_PI / radar_echo.step_angle.getRad()) ||
+                m_current_range != radar_echo.range) {
+                m_current_sweep_size = radar_echo.sweep_length;
+                m_current_num_angles = 2 * M_PI / radar_echo.step_angle.getRad();
+                m_current_range = radar_echo.range;
+                updateLookUpTable();
+            }
+            addEchoesToFrame(radar_echo, yaw_correction);
+        } while (_echo.read(radar_echo, false) == RTT::NewData);
+
+        if (base::Time::now() - m_last_sample >
+            _export_config.get().time_between_frames) {
```

Comment:
I think using deadlines makes understanding time related code easier. Also, create an instance variable for the export config, you `.get()` too many times which makes the code noisy.